src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenByAuthorizationCodeHandler.cs

```diff
@@ -69,9 +70,9 @@ protected override void AddAditionalRequestParameters(DictionaryRequestParameter
             requestParameters[OAuthParameter.RedirectUri] = this.redirectUri.OriginalString;
         }
 
-        protected override void PostTokenRequest(AuthenticationResultEx resultEx)
+        protected override async Task PostTokenRequest(AuthenticationResultEx resultEx)
```

Comment:
because we call asynchronous method UpdateFromTemplateAsync in PostTokenRequest, when broker returns Authority in resultEx.Result and  we want to update it in  Authenticator